lib/spdy-transport/priority.js

```diff
@@ -71,10 +71,13 @@ PriorityNode.prototype.remove = function remove () {
 PriorityNode.prototype.removeChild = function removeChild (child) {
   this.children.weight -= child.weight
   var index = utils.binarySearch(this.children.list, child, compareChildren)
-  assert(index !== -1)
-
-  // Remove the child
-  this.children.list.splice(index, 1)
+  try {
```

Comment:
Not an author or reviewer but I'm stuck with this issue as well. I just noticed this logic could be done a bit easier and without the `try catch`:

```js
if (index !== -1 && this.children.list.length >= index) {
  this.children.list.splice(index, 1);
}
```